cdap-ui/app/directives/dag/my-dag-ctrl.js

```diff
@@ -169,13 +175,23 @@ angular.module(PKG.name + '.commons')
             angular.forEach(labels, function (endpoint) {
               var label = endpoint.getOverlay('metricLabel');
               if ($scope.metricsData[endpoint.elementId] === null || $scope.metricsData[endpoint.elementId] === undefined) {
-                angular.element(label.getElement())
+                angular.element(label.getElement()).children()
                   .text(0);
                 return;
               }
 
-              angular.element(label.getElement())
-                .text(numberFilter($scope.metricsData[endpoint.elementId].recordsOut, 0));
+              var recordsOut = $scope.metricsData[endpoint.elementId].recordsOut;
+
+              // hide label if the length is greater than 12 characters
```

Comment:
because it's an integer, I can just check if it's greater than that number. I check for that on line 186